src/plugins/piral-modals/src/Modals.tsx

```diff
@@ -21,9 +21,9 @@ function findModal(modals: Record<string, ModalRegistration>, name: string): Mod
 
 export const Modals: React.FC = () => {
   const modals = useGlobalState((s) => s.registry.modals);
-  const dialogs = useGlobalState((s) => s.modals);
+  const dialogs = [...useGlobalState((s) => s.modals)];
```

Comment:
I think there is no need to change this line.

Just place `.reverse()` after the `filter(Boolean). This way, we have an allocation less and perform the re-sorting on a potentially smaller array.